controllers/runner_controller.go

```diff
@@ -874,30 +874,44 @@ func newRunnerPod(template corev1.Pod, runnerSpec v1alpha1.RunnerConfig, default
 	// When you're NOT using dindWithinRunner=true,
 	// it must also be shared with the dind container as it seems like required to run docker steps.
 	//
+	// Setting VolumeSizeLimit to zero will disable /runner emptydir mount
+	//
+	// Enabling VolumeUseMemory will use tmpfs (RAM-backed filesystem) for emptydirs
+	//
 
 	runnerVolumeName := "runner"
 	runnerVolumeMountPath := "/runner"
-	runnerVolumeEmptyDir := &corev1.EmptyDirVolumeSource{}
+	runnerVolumeEmptyDirMedium := corev1.StorageMediumDefault
+
+	if runnerSpec.VolumeUseMemory != nil && *runnerSpec.VolumeUseMemory {
+		runnerVolumeEmptyDirMedium = corev1.StorageMediumMemory
+	}
```

Comment:
```suggestion
	if runnerSpec.VolumeMedium != nil {
		runnerVolumeEmptyDirMedium = *runnerSpec.VolumeMedium
	}
```

based on https://github.com/actions-runner-controller/actions-runner-controller/pull/674/files#r667401048